pkg/webhook/validator.go

```diff
@@ -1012,3 +1038,104 @@ func refOrFieldError(image, field string, index int) (name.Reference, *apis.Fiel
+func getConfigs(ctx context.Context, ref name.Reference, options ...remote.Option) (map[string]*v1.ConfigFile, []error) {
+	descriptor, err := remote.Get(ref, options...)
+	if err != nil {
+		return nil, []error{fmt.Errorf("failed to get ref %s : %w", ref.String(), err)}
+	}
+	switch descriptor.MediaType {
+	case types.OCIImageIndex, types.DockerManifestList:
+		ii, err := descriptor.ImageIndex()
+		if err != nil {
+			return nil, []error{fmt.Errorf("getting ImageIndex for %s : %w", ref.String(), err)}
+		}
+		im, err := ii.IndexManifest()
+		if err != nil {
+			return nil, []error{fmt.Errorf("getting IndexManifest for %s : %w", ref.String(), err)}
+		}
+		wg := new(sync.WaitGroup)
+
+		results := make(chan configFileResult, len(im.Manifests))
+		for _, manifest := range im.Manifests {
+			manifest := manifest
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+
+				newRefString := fmt.Sprintf("%s/%s@%s", ref.Context().RegistryStr(), ref.Context().RepositoryStr(), manifest.Digest)
+				newRef, err := name.ParseReference(newRefString)
+				if err != nil {
+					results <- configFileResult{ret: nil, errs: []error{fmt.Errorf("failed to ParseReference for: %s: %w", newRefString, err)}}
+					return
+				}
+
+				newRefConfigs, errs := getConfigs(ctx, newRef)
+				results <- configFileResult{ret: newRefConfigs, errs: errs}
+			}()
+		}
+		errs := []error{}
+		ret := make(map[string]*v1.ConfigFile, len(im.Manifests))
+		for i := 0; i < len(im.Manifests); i++ {
+			select {
+			case <-ctx.Done():
+				errs = append(errs, errors.New("context canceled"))
+			case result, ok := <-results:
+				if !ok {
+					errs = append(errs, errors.New("channel closed before all results were gathered"))
+				} else {
+					if result.errs != nil {
+						errs = append(errs, fmt.Errorf("failed to get a ConfigFile: %w", err))
+					} else {
+						for k, v := range result.ret {
+							ret[k] = v
+						}
+					}
+				}
+			}
+		}
+		wg.Wait()
+		if len(errs) > 0 {
+			return nil, errs
+		}
+		return ret, nil
+	case types.OCIManifestSchema1, types.DockerManifestSchema2:
+		// This is an Image, so just return it.
+		image, err := descriptor.Image()
+		if err != nil {
+			return nil, []error{fmt.Errorf("getting Image for %s: %w", ref.String(), err)}
+		}
+		cf, err := image.ConfigFile()
+		if err != nil {
+			return nil, []error{fmt.Errorf("getting ConfigFile for %s: %w", ref.String(), err)}
+		}
+		return map[string]*v1.ConfigFile{normalizeArchitecture(cf): cf}, nil
+	default:
+		return nil, []error{fmt.Errorf("unknown mime type for %s: %v", ref.String(), descriptor.MediaType)}
+	}
+}
+
+// normalizeArchitecture normalizes the os/architecture/variant to:
+// {OS}/{Architecture}[/{Variant}]
+//
+// Some examples are:
+// linux/arm64
+// linux/arm/v7
+// linux/arm/v6
+func normalizeArchitecture(cf *v1.ConfigFile) string {
+	if cf.Variant != "" {
+		return fmt.Sprintf("%s/%s/%s", cf.OS, cf.Architecture, cf.Variant)
+	}
+	return fmt.Sprintf("%s/%s", cf.OS, cf.Architecture)
+}
```

Comment:
Consider constructing a `v1.Platform` and using: https://github.com/google/go-containerregistry/blob/bdc946f4d482443524054e63fc56c11811858cae/pkg/v1/platform.go#L33

I think you get these for free from the multi-arch variants: https://github.com/google/go-containerregistry/blob/bdc946f4d482443524054e63fc56c11811858cae/pkg/v1/manifest.go#L49